dan_layer/storage/src/consensus_models/block.rs

```diff
@@ -293,12 +317,20 @@ impl Block {
             timestamp: parent_timestamp,
             base_layer_block_height: parent_base_layer_block_height,
             base_layer_block_hash: parent_base_layer_block_hash,
+            extra_data: None,
         };
         block.id = block.calculate_hash().into();
         block.is_justified = false;
         block
     }
 
+    fn extra_data_from_sidechain_id(sidechain_id: Option<RistrettoPublicKey>) -> Result<Option<ExtraData>, BlockError> {
+        let extra_data = sidechain_id
+            .map(|id| ExtraData::new().insert_sidechain_id(id).cloned())
+            .transpose()?;
+        Ok(extra_data)
+    }
+
     pub fn calculate_hash(&self) -> FixedHash {
```

Comment:
extra fields needs to go into the hash (BlockId) to prevent malleability and to commit to the genesis sidechain id.